### PR TITLE
fix spi oauth image build

### PR DESCRIPTION
**Pull request by albarbaro**
Fix setup for service-provider-oauth image build in e2e jobs.

### CODE REVIEW

**Comment by psturc** [comment]
I'm not sure if the change in e2e step is compatible with other e2e than SPI. let's double check that just to be sure
/hold
/pj-rehearse pull-ci-redhat-appstudio-release-service-main-release-service-e2e

File: ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-ref.yaml [hunk 1]
```diff
@@ -16,6 +16,8 @@ ref:
     name: "release:latest"
   - env: CI_JBS_REQPROCESSOR_IMAGE
     name: "release:latest"
+  - env: CI_SPI_OAUTH_IMAGE
+    name: "service-provider-integration-oauth"
```
Comment (psturc) [inline ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-ref.yaml:20]: I'm not sure if this is allowed. IMO the name of the image specified in the step must be compatible with an image name generated for every other job, e.g. "release:latest". but I might be wrong. let's double check that with pj-rehearse

Comment (psturc) [reply ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-ref.yaml:20]: @albarbaro see [this failure](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/45433/rehearse-45433-pull-ci-redhat-appstudio-release-service-main-release-service-e2e/1722249545573732352#1:build-log.txt%3A45)
```suggestion
    name: "release:latest"
```

Comment (albarbaro) [reply ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-ref.yaml:20]: you were right, it failed. what do you suggest to put there?

File: ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-ref.yaml [hunk 2]
```diff
@@ -16,6 +16,8 @@ ref:
     name: "release:latest"
   - env: CI_JBS_REQPROCESSOR_IMAGE
     name: "release:latest"
+  - env: CI_SPI_OAUTH_IMAGE
+    name: "release:latest"
```
Comment (flacatus) [inline ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-ref.yaml:20]: why not redhat-appstudio-spi-oauth-image? and also i dont understand why CI_JBS_REQPROCESSOR_IMAGE

**Review by flacatus — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-ref.yaml (hunk 2 above)
Comment (psturc) [reply ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-ref.yaml:20]: [explanation](https://github.com/redhat-appstudio/e2e-tests/blob/dd757fc91f25eef1daf4d453785424a3fec60670/magefiles/magefile.go#L332-L337)

Comment (flacatus) [reply ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-ref.yaml:20]: 💥 head

**Review by psturc — APPROVED** [review]
/lgtm

**Pull request merged**